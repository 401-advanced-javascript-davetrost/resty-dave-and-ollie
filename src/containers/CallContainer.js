import React from 'react';
import CallForm from '../components/CallForm';
import Response from '../components/Response';
import PropTypes from 'prop-types';

export default function CallContainer({ url, method, body, response, handleUrlChange, handleMethodChange, handleBodyChange, handleSubmit }) {
  return (
    <section>
      <CallForm 
        url={url} 
        method={method} 
        body={body}
        handleUrlChange={handleUrlChange} 
        handleMethodChange={handleMethodChange} 
        handleBodyChange={handleBodyChange} 
        handleSubmit={handleSubmit}/>
      <Response 
        response={response}/>
    </section>
  );
}

CallContainer.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  response: PropTypes.string,
  handleUrlChange: PropTypes.func.isRequired,
  handleMethodChange: PropTypes.func.isRequired,
  handleBodyChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};
