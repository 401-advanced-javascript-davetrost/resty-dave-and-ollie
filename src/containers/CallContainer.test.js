import React from 'react';
import { shallow } from 'enzyme';
import CallContainer from './CallContainer';

describe('Call Container component', () => {
  it('renders the container', () => {
    const responseExampleString = `[
      {
        "_id": "5da237699734fdcb7bef8f56",
        "name": "Miles Shortman",
        "image": "https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
      }
    ]`;
    const wrapper = shallow(<CallContainer 
      url="random-api.com/api/characters"
      method="get"
      body=""
      response={responseExampleString}
      handleUrlChange={() => true}
      handleMethodChange={() => true}
      handleBodyChange={() => true}
      handleSubmit={() => true}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
