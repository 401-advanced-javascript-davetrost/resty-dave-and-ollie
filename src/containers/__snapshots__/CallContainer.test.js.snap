// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Call Container component renders the container 1`] = `
<section>
  <CallForm
    body=""
    handleBodyChange={[Function]}
    handleMethodChange={[Function]}
    handleSubmit={[Function]}
    handleUrlChange={[Function]}
    method="get"
    url="random-api.com/api/characters"
  />
  <Response
    response="[
      {
        \\"_id\\": \\"5da237699734fdcb7bef8f56\\",
        \\"name\\": \\"Miles Shortman\\",
        \\"image\\": \\"https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639\\"
      }
    ]"
  />
</section>
`;
